plutus-core/plutus-core/src/PlutusCore/Pretty/Readable.hs

```diff
@@ -72,9 +89,78 @@ botPrettyConfigReadable configName = PrettyConfigReadable configName botRenderCo
 topPrettyConfigReadable :: configName -> ShowKinds -> PrettyConfigReadable configName
 topPrettyConfigReadable configName = PrettyConfigReadable configName topRenderContext
 
+-- | The fixity of a binder.
+binderFixity :: Fixity
+binderFixity = Fixity RightAssociative 1
+
+-- | The fixity of @(->)@.
+arrowFixity :: Fixity
+arrowFixity = Fixity RightAssociative 2
+
 -- | Pretty-print two things with a @->@ between them.
 arrowPrettyM
     :: (MonadPrettyContext config env m, PrettyBy config a, PrettyBy config b)
     => a -> b -> m (Doc ann)
 arrowPrettyM a b =
     infixDocM arrowFixity $ \prettyL prettyR -> prettyL a <+> "->" <+> prettyR b
+
+iterTyForallPrettyM
+    :: ( MonadPrettyContext config env m
+       , PrettyBy config arg, PrettyBy config body
+       )
+    => [arg] -> body -> m (Doc ann)
+iterTyForallPrettyM args body =
+    compoundDocM binderFixity $ \prettyIn ->
+        let prettyBot x = prettyIn ToTheRight botFixity x
+            prettyBinds = align . vsep $ map (prettyIn ToTheLeft binderFixity) args
+        in ("all" <+> prettyBinds <> ".") <?> prettyBot body
+
+iterLamAbsPrettyM
+    :: ( MonadPrettyContext config env m
+       , PrettyBy config arg, PrettyBy config body
+       )
+    => [arg] -> body -> m (Doc ann)
+iterLamAbsPrettyM args body =
+    -- Lay out abstraction like
+    --
+    --     \(x : t)
+    --      (y : t')
+    --      (z : t'') ->
+    --       body
+    compoundDocM binderFixity $ \prettyIn ->
+        let prettyBot x = prettyIn ToTheRight botFixity x
+            prettyBinds = align . vsep $ map (prettyIn ToTheLeft binderFixity) args
+        in ("\\" <> prettyBinds <+> "->") <?> prettyBot body
+
+iterTyAbsPrettyM
+    :: ( MonadPrettyContext config env m
+       , PrettyBy config arg, PrettyBy config body
+       )
+    => [arg] -> body -> m (Doc ann)
+iterTyAbsPrettyM args body =
```

Comment:
Why is this different to `iterLamAbsPrettyM`? I would expect it to be identical except for the initial `/\` rather than `\`?

Reply:
Great question! I've no idea, those were written by Max. I agree they should probably look the same and I'll try unifying them.